doc/secure_boot/hld_secure_boot.md

```diff
@@ -0,0 +1,383 @@
+/boot/efi/EFI location: Basically when ONIE installed is creates a boot general partition.
+So It is required to mount this created partition with this name /boot/efi/EFI and copy the shim file to this location.
+
+- So, when the boot process starts, UEFI will look for the shim file in this location to verify it and then continue with the boot flow.
+- There is a possibility to download shim already signed by Microsoft, but then, if we need to do some modifications, or choose another Shim version in the future we will depend on having a version that Microsoft supports as well.
+
+##### GRUB SONiC modification
+Currently, the Grub file is created when installing a SONiC image, the code is in the prefix of the SONiC image that contained the script install.sh and by using the tool:
+    
+    grub-install
+
+The problem with this approach is that in SB we need the grub to be signed, so the grub EFI file should be created in build flow and signed instead of creating it in install.sh.
+
+In order to have the grubx64.efi/grubarm.efi in the build flow we are going to install a Debian package name:
+
+    grub-efi
+
+The output of this installation is the file grubx64.efi, after the creation, it will be signed and copied to the path: /boot/.
+
+Note: In case the build is without SB the grub flow will not be changed as the original.
+
+##### Vmlinuz SONiC modification
+In SONiC the submodule name sonic-linux-kernel contained the Linux kernel.
+
+We need to do the following modification to support SB:
+- We need to integrate the pub key that will be used to verify kernel modules, to do that, we need the following modification in the Linux kernel configuration:
+
+- CONFIG_MODULE_SIG_FORCE=y
+
+enforce to check the signature of KO.
+
+- CONFIG_MODULE_SIG_KEY=""
+
+ this is a configuration to enforce vmlinuz to sign kernel modules, but we will do that differently, so it will be empty (see explanation in Kernel Modules verification).
+
+- CONFIG_SYSTEM_TRUSTED_KEYS=< custom_vendor_certificates_path > (Format file=pem)
+
+this variable is saying where the public key is. This key will verify the kernel modules, this path should be custom chosen by the user.
+
+So, we will add a build variable where the developer can set the key location in his environment.
+
+- CONFIG_KEXEC_VERIFY_SIG=y
+Details in Warmboot chapter.
+
+Finally, after the configuration changes, we need to sign the file vmlinuz.efi (similar like we sign shim, grub with the sign script)
+
+Note: Same as other modifications if the SB_BUILD flag is “no sign” those modifications will not happen and the kernel will remain the same as before the feature was implemented
+##### Kernel Module SONiC modification
+Most of the kernel modules are in sonic-linux-kernel repo, but there are more KO like Ofed & SDK.
+
+The secure_boot_sign.py script wrapper the sign-file tool as well and will sign all the KO.
+
+All the KO can be found in fsroot-mellanox/ directory and will be signed at the end of the build.
```

Comment:
why are we limiting this to only mellanox directory?

Reply:
fix the named